src/components/Nav/AddBook/index.js

```diff
@@ -0,0 +1,177 @@
+import { useState } from "react";
+import "./add-book.css";
+import { Navigate } from "react-router-dom";
+
+function AddBook() {
+  const [title, setTitle] = useState("");
+  const [author, setAuthor] = useState("");
+  const [genre, setGenre] = useState("");
+  const [year, setYear] = useState(0);
+  const [pageCount, setPageCount] = useState("");
+  const [imgUrl, setImgUrl] = useState("");
+  const [blurb, setBlurb] = useState("");
+
+  const [redirect, setRedirect] = useState(false);
+
+  const [titleErrors, setTitleErrors] = useState("");
+  const [authorErrors, setAuthorErrors] = useState("");
+  const [genreErrors, setGenreErrors] = useState("");
+
+  function changeTitle(event) {
+    setTitle(event.target.value);
+  }
+  function changeAuthor(event) {
+    setAuthor(event.target.value);
+  }
+  function changeGenre(event) {
+    setGenre(event.target.value);
+  }
+  function changeYear(event) {
+    setYear(event.target.value);
+  }
+  function changePageCount(event) {
+    setPageCount(event.target.value);
+  }
+  function changeImgUrl(event) {
+    setImgUrl(event.target.value);
+  }
+  function changeBlurb(event) {
+    setBlurb(event.target.value);
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const postData = {
+      "title": title,
+      "author": author,
+      "genre_id": genre,
+    };
+
+    if (blurb.length != 0) {
+      postData.blurb = blurb;
+    }
+
+    if (imgUrl.length != 0) {
+      postData.image = imgUrl;
+    }
+
+    if (year != 0) {
+      postData.year = year;
+    }
+
+    if (pageCount != 0) {
+      postData.page_count = pageCount;
+    }
```

Comment:
I would consider moving this into a function `generatePostData` or something just to keep each function as small as possible